flocker/volume/service.py

```diff
@@ -372,13 +384,28 @@ def handoff(self, volume, destination):
             errbacks on error (specifcally with a ``ValueError`` if the
             volume is not locally owned).
         """
-        pushing = maybeDeferred(self.push, volume, destination)
-
-        def pushed(ignored):
-            remote_uuid = destination.acquire(volume)
-            return volume.change_owner(remote_uuid)
-        changing_owner = pushing.addCallback(pushed)
-        return changing_owner
+        compute_driver, volume_driver = driver_from_environment()
+        from subprocess import check_call
+        # unmount volume
+        volume_mount_path = volume.get_filesystem().get_path()
+        check_call(['umount', volume_mount_path.path])
+        FilePath.remove(volume_mount_path)
```

Comment:
Cleaning up the unused mount directory seems like a good idea. But discuss the use of the node_id in the mountpoint directory name, which seems to be used only in support of zfs send / recv between nodes.